src/IB/FEMechanicsExplicitIntegrator.cpp

```diff
@@ -442,6 +440,240 @@ FEMechanicsExplicitIntegrator::modifiedTrapezoidalStep(const double current_time
+                               P_current_vec,
+                               F_vec.get(),
+                               dP_dt_vec.get(),
+                               current_time,
+                               current_time + 1.0 * dt,
+                               part);
+
+            doForwardEulerStep(X2_vec.get(),
+                               U2_vec.get(),
+                               P2_vec.get(),
+                               X1_vec,
+                               U1_vec,
+                               P1_vec,
+                               F_vec.get(),
+                               dP_dt_vec.get(),
+                               current_time + 1.0 * dt,
+                               current_time + 2.0 * dt,
+                               part);
+
+            ierr = VecAXPBY(U2_vec->vec(), 0.75, 0.25, U_current_vec->vec());
+            IBTK_CHKERRQ(ierr);
+            ierr = VecAXPBY(X2_vec->vec(), 0.75, 0.25, X_current_vec->vec());
+            IBTK_CHKERRQ(ierr);
+            if (d_dynamic_pressure_part[part])
+            {
+                ierr = VecAXPBY(P2_vec->vec(), 0.75, 0.25, P_current_vec->vec());
+                IBTK_CHKERRQ(ierr);
+            }
+
+            doForwardEulerStep(X_new_vec,
+                               U_new_vec,
+                               P_new_vec,
+                               X2_vec.get(),
+                               U2_vec.get(),
+                               P2_vec.get(),
+                               F_vec.get(),
+                               dP_dt_vec.get(),
+                               current_time + 0.5 * dt,
+                               current_time + 1.5 * dt,
+                               part);
+
+            ierr = VecAXPBY(U_new_vec->vec(), 1.0 / 3.0, 2.0 / 3.0, U_current_vec->vec());
+            IBTK_CHKERRQ(ierr);
+            ierr = VecAXPBY(X_new_vec->vec(), 1.0 / 3.0, 2.0 / 3.0, X_current_vec->vec());
+            IBTK_CHKERRQ(ierr);
+            if (d_dynamic_pressure_part[part])
+            {
+                ierr = VecAXPBY(P_new_vec->vec(), 1.0 / 3.0, 2.0 / 3.0, P_current_vec->vec());
+                IBTK_CHKERRQ(ierr);
+            }
+
+            d_F_vecs->get("new", part) = *F_vec;
+        }
+        else if (n_stages == 4)
+        {
+            // We can avoid allocating vectors for X1, U1, and P1 because these vectors are never used with X_new,
+            // U_new, or P_new.
+            PetscVector<double>* U_current_vec = &d_U_vecs->get("current", part);
+            PetscVector<double>* U_new_vec = &d_U_vecs->get("new", part);
+            PetscVector<double>* U1_vec = U_new_vec;
+            std::unique_ptr<PetscVector<double> > U2_vec(
+                dynamic_cast<PetscVector<double>*>(d_U_vecs->get("new", part).clone().release()));
+            std::unique_ptr<PetscVector<double> > U3_vec(
+                dynamic_cast<PetscVector<double>*>(d_U_vecs->get("new", part).clone().release()));
```

Comment:
```suggestion
            PetscVector<double> &U2 = d_U_vecs->get("U2", part);
            PetscVector<double> &U3 = d_U_vecs->get("U3", part);
```

we can let `d_U_vecs` et al do the indexing and cloning for us. We shouldn't need to use `clone().release()` anywhere. An alternative, which creates vectors outside of the `System`, would be

```cpp
const auto &comm = vec.comm();
PetscVector<double> U2(comm), U3(comm), X_current(comm), X_new(comm);
U2 = U;
U3 = U;
X_current = X;
X_new = X;
```
since `operator=` is defined but copy constructors are not.

Reply:
Oh yeah, I forgot about how these are set up!